analyzer/tests/unit/test_option_parser.py

```diff
@@ -229,8 +229,10 @@ def test_preprocess_and_compile(self):
     def test_ignore_flags(self):
         """
         Test if special compiler options are ignored properly.
+        In case the compiler is clang then the original command should be
```

Comment:
I think we should create a separate test case for not modifying clang compilation actions.

With this change you modify the tests for the gcc/g++ case and remove the information that the ` "-MT hello", "-M",` flags are filtered out in that case.

Reply:
These precompilation related flags are filtered in case of clang too.